ibm-build-lab/VAD-VAR-Workshop
# feat: Localization added

Content changes:

- Test page has two versions to show how locale switching works in dev

Code changes:

- Localization added. Below's a brief summary of the implementation
    -  Content for each locale is stored in seperate MD files as `filename.[locale].md`. If no locale is present the file is assumed to be in English
    - When static files are parsed at build time their slugs are created with the locale prepended at the beginning. For instance, `/watson/README.es.md` would become `/es/watson`
    - `react-i18next` has been added as well to facilitate locale state
    - The left size nav is built based on current locale. If the locale is changed the side nav will update to show the pages available in that locale
    - The `LanguageSelector` in the top right exist to allow for changes in locale. It will check if the current page supports multiple locales and if it does the dropdown will not be disabled. Switching the locale just navigates to another static page.
- Notes on writing content
   - Relative links do not need locales add. The `SmartLink` component sources the current locale and prepends it to the relative link. For instance, a link like `/watson/101` as written in the MD would become `[locale]/watson/101`.
   - Files for various locales live next to each other so image tags paths remain the same in Markdown


Testing:

Currently the only multi-locale page is the test page in the dev environment. 

Steps:
1. Start the local dev server with `npm run dev`
2. Navigate to the test page and try switching the locale to see the side nav update
